ark/phenotyping/cell_consensus_cluster.R

```diff
@@ -18,40 +20,59 @@ library(ConsensusClusterPlus)
 # get the command line arguments
 args <- commandArgs(trailingOnly=TRUE)
 
+# get the cluster cols to subset over
+pixelClusterCol <- args[1]
+
 # get number of consensus clusters
-maxK <- strtoi(args[1])
+maxK <- strtoi(args[2])
 
 # get z-score scaling factor
-cap <- strtoi(args[2])
+cap <- strtoi(args[3])
 
 # get the cell cluster path
-cellClusterPath <- args[3]
+cellClusterPath <- args[4]
 
 # get path to the averaged cluster data
-clusterAvgPath <- args[4]
+clusterAvgPath <- args[5]
 
 # get consensus cluster write path
-cellConsensusPath <- args[5]
+cellConsensusPath <- args[6]
+
+# get the clust to meta write path
+clustToMeta <- args[7]
 
 # set the random seed
-seed <- strtoi(args[6])
+seed <- strtoi(args[8])
 set.seed(seed)
 
 print("Reading cluster averaged data")
-clusterAvgs <- arrow::read_feather(clusterAvgPath)
+clusterAvgs <- as.data.frame(read.csv(clusterAvgPath, check.names=FALSE))
 
 # scale and cap the data respectively
+# NOTE: z-scoring and capping cluster avg data produces better clustering results
 print("Scaling data")
-clusterAvgsScale <- pmin(scale(clusterAvgs), cap)
+clusterCols <- colnames(clusterAvgs)[grepl(pattern=sprintf('%s_', pixelClusterCol),
+                                     colnames(clusterAvgs))]
+clusterAvgsScale <- pmin(scale(clusterAvgs[,clusterCols]), cap)
```

Comment:
Like in pixel clustering, I think you are z-scoring twice. Once here, and once in compute_cell_cluster_count_avg.